tests/MongoDB.Driver.Tests/OcspIntegrationTests.cs

```diff
@@ -0,0 +1,139 @@
+
+        public OcspIntegrationTests(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        /*
+         * This test must be run as its own task in Evergreen.
+         * When testing locally, this test should not run alongside other integration tests.
+         * Clearing local OCSP caches between test runs is also required when testing locally.
+         * Clearing the OCSP caches may involve clearing out the caches in ~\.dotnet folder
+         * and $DOTNET_CLI_HOME/.dotnet, as well as running `dotnet clean` prior to running the tests.
+         * See spec for OCSP OS-level cache manipulation commands.
+         *   https://github.com/mongodb/specifications/blob/master/source/ocsp-support/ocsp-support.rst#os-level-ocsp-cache-manipulation
+         * When testing on Windows, the certificate should be added to the trust store prior to each run in order to
+         * reduce the chances of Windows pruning the certificate from the trust store prior to the test running.
+         */
+        [SkippableFact]
+        public void MongoClientShouldRespectCertificateStatusAndTlsInsecure()
+        {
+            /* We cannot call RequireServer.Check() because this would result in a connection being made to the mongod
+             * and that connection will not succeed if we're testing the revoked certificate case. */
+
+            RequireEnvironment.Check().EnvironmentVariable(_shouldSucceedEnvironmentVariableName);
+
+            // Uncommenting the line below may help when testing locally
+            // SetupLocalTestingEnvironmentVariables(mockResponderValidatesAllCertificates: false);
+            var shouldSucceed = GetShouldSucceed();
+            /* To prevent OCSP caching from polluting the test results, we MUST run the "secure" version before
+             * the tlsInsecure version.  */
+            var secureClientException = Record.Exception(() => Ping(tlsInsecure: false));
+            var tlsInsecureClientException = Record.Exception(() => Ping(tlsInsecure: true));
+
+            tlsInsecureClientException.Should().BeNull();
+            if (shouldSucceed)
+            {
+                secureClientException.Should().BeNull();
+            }
+            else
+            {
+                secureClientException.Should().BeOfType<TimeoutException>();
+                var message = secureClientException.Message;
+                // The exception will lack this message if the heartbeat doesn't fire
+                message.Should().Contain("The remote certificate is invalid according to the validation procedure.");
+            }
+
+            void Ping(bool tlsInsecure)
+            {
+                using (var client = CreateDisposableMongoClient(tlsInsecure))
+                {
+                    client.GetDatabase("admin").RunCommand<BsonDocument>(new BsonDocument("ping", 1));
+                    if (client.Settings.SdamLogFilename != null)
```

Comment:
As I understand, you commented out specifying `settings.SdamLogFilename`, but this line is still presented. So now this functionality is not used at all?

Reply:
I agree that it can be helpful for local debugging (as well as for a lot of other tests), but I would prefer to have only code that is going to be executed on the server as we usually do. As an option, the description of steps for local debugging can be added to comments before the test.

Reply:
You're right that this code is currently inactive. However, this functionality is needed in the case that someone desires to turn sdam logging: it's also entirely non obvious that the client cannot be disposed early in order for the logging to work: it took me quite a while to figure this out when testing locally. In this case, in the interest of time, I will leave this code in unless you strongly disagree.